internal/controller/postgrescluster/pgbouncer_test.go

```diff
@@ -365,3 +371,87 @@ func TestReconcilePGBouncerDeployment(t *testing.T) {
 	})
 
 }
+
+func TestReconcilePGBouncerDisruptionBudget(t *testing.T) {
+	env, cc, config := setupTestEnv(t, ControllerName)
+	t.Cleanup(func() { teardownTestEnv(t, env) })
+
+	r := &Reconciler{}
+	ctx, cancel := setupManager(t, config, func(mgr manager.Manager) {
+		r = &Reconciler{
+			Client:   cc,
+			Owner:    client.FieldOwner(t.Name()),
+			Recorder: mgr.GetEventRecorderFor(ControllerName),
+			Tracer:   otel.Tracer(t.Name()),
+		}
+	})
+	t.Cleanup(func() { teardownManager(cancel, t) })
+
+	foundPDB := func(
+		t *testing.T,
+		cluster *v1beta1.PostgresCluster,
+		minAvailable int32,
+	) bool {
+		pdb, err := r.generatePodDisruptionBudget(
+			cluster,
+			naming.ClusterPGBouncer(cluster),
+			minAvailable,
+			naming.ClusterPGBouncerSelector(cluster),
+		)
+		assert.NilError(t, err)
+		got := &policyv1beta1.PodDisruptionBudget{}
+		err = r.Client.Get(ctx, client.ObjectKeyFromObject(pdb), got)
```

Comment:
⛏️ `pdb` is used only for its `ObjectKey`, but we're also building/passing the name on L397. Can the call to `generatePodDisruptionBudget` be replaced with the following?

```suggestion
		err = r.Client.Get(ctx, naming.AsObjectKey(naming.ClusterPGBouncer(cluster)), got)
```